singlefactorauth/src/main/java/com/web3auth/singlefactorauth/SingleFactorAuth.kt

```diff
@@ -20,46 +25,62 @@ import org.torusresearch.torusutils.types.VerifierParams
 import org.torusresearch.torusutils.types.VerifyParams
 import org.torusresearch.torusutils.types.common.TorusKey
 import org.torusresearch.torusutils.types.common.TorusOptions
-import org.torusresearch.torusutils.types.common.TorusPublicKey
 import org.web3j.crypto.Hash
 import java.util.concurrent.CompletableFuture
 
 class SingleFactorAuth(
-    sfaParams: SFAParams,
+    web3AuthOptions: Web3AuthOptions,
     ctx: Context,
 ) {
     private var nodeDetailManager: FetchNodeDetails =
-        FetchNodeDetails(sfaParams.getNetwork())
+        FetchNodeDetails(web3AuthOptions.getNetwork())
     private val torusUtils: TorusUtils
     private var sessionManager: SessionManager
     private val gson = GsonBuilder().disableHtmlEscaping().create()
     private var network: Web3AuthNetwork
+    private var state: SessionData? = null
 
     init {
         val torusOptions = TorusOptions(
-            sfaParams.getClientId(), sfaParams.getNetwork(),
-            sfaParams.getLegacyMetaDataHostUrl(), sfaParams.getServerTimeOffset(), true
+            web3AuthOptions.getClientId(), web3AuthOptions.getNetwork(), null,
+            web3AuthOptions.getServerTimeOffset(), true
         )
-        network = sfaParams.getNetwork()
+        network = web3AuthOptions.getNetwork()
         torusUtils = TorusUtils(torusOptions)
-        sessionManager = SessionManager(ctx, sfaParams.getSessionTime(), ctx.packageName)
+        sessionManager = SessionManager(ctx, web3AuthOptions.getSessionTime(), ctx.packageName)
     }
 
-    fun initialize(ctx: Context): CompletableFuture<SFAKey> {
-        val sfaCF = CompletableFuture<SFAKey>()
-        val data = sessionManager.authorizeSession(ctx.packageName, ctx)
-        data.whenComplete { response, error ->
-            val mainHandler = Handler(Looper.getMainLooper())
-            mainHandler.post {
-                if (error != null) {
-                    sfaCF.completeExceptionally(error)
-                } else {
-                    sfaCF.complete(
-                        gson.fromJson(
-                            JSONObject(response).toString(),
-                            SFAKey::class.java
+    fun initialize(ctx: Context): CompletableFuture<SessionData> {
```

Comment:
Simplify this to: ```fun initialize(ctx: Context): SessionData```

The completablefuture itself can also be simplified, using supplyasync({}), then get() for the result.

Consider also using future instead of completablefuture.